File: coding/sieveOfEratosthenes.java
class sieveOfEratosthenes
{

public static void main(String args[])
{
	long  range;
	long  currentNum;
	long  multiple;
	long  count;
	long  prime;
	long  prev;
	long  inc;
	range = 10000000;
	boolean arr[]=new long long[range+1];
	arr[1]=1;
	arr[2]=1;
	arr[0]=1;
	count = 1 ;
	for(multiple = 2; multiple < range ; multiple+=2)
	{
		arr [ multiple ] = 1;
		count++;
	}
	prime = 3;
	
	while(prime < range)
	{
		prev = prime;
		while( prime < range && prime [arr]  )
		{
			prime+=2;
		}
//		cout << " next Selected prime :" <<prime;
		inc = prime * 2;
		for(multiple = prime * prime ; multiple < range ; multiple += inc )
		{
			if( ! arr [ multiple ]) count++;
			arr [ multiple ] = 1 ;
			
		}
		prev [ arr ] = 0;
		prime [ arr ] = 1;
	}
//	for(multiple=2;multiple<range; multiple+=1)
//	{
//		if(!arr [ multiple ])
//		{
//			cout<< multiple << "\n" ;
//			count+=1;
//		}
//	}
	cout << "Count : " << (range - count);
}
}

File: coding/others/sieveOfEratosthenes.cpp
// prime number in a range
// Sieve of Eratosthenes 
/*
 * Count of prime numbers below 10: 4
Count of prime numbers below 100: 25
Count of prime numbers below 1,000: 168
Count of prime numbers below 10,000: 1,229
Count of prime numbers below 100,000: 9,592
Count of prime numbers below 1,000,000: 78,498
Count of prime numbers below 10,000,000: 664,579
 */
#include<iostream>
#include <stdlib.h>
using namespace std;
int main()
{
	long long  range;
        long long  multiple;
        long long  count;
        long long int prime;
        long long inc;
        range = 1000000;
        bool arr[range+1]={true};
        arr[1]=true;
        arr[2]=true;
        arr[0]=true;
        count = 1 ;
        for(multiple = 2; multiple < range ; multiple+=2)
        {
            arr [ multiple ] = true;
            count++;
        }
        prime = 3;
        // prime * prime < range not only ( prime < range ) 
        while(prime * prime < range)
        {
            if(!arr[prime]){
                inc = prime * 2;
                for(multiple = prime * prime ; multiple < range; multiple += inc )
                {
                    if( ! arr [ multiple ]) count++;
                    arr [ multiple ] = true ;
                    
                }
            }
            prime+=2;

        }
		cout << "Count : " << (range - count);

	return EXIT_SUCCESS;
}

